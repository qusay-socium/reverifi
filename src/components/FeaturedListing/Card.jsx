import React from 'react';
import PropTypes from 'prop-types/prop-types';

import colors from 'styles/colors';
import {
  CardBody,
  CardContainer,
  Flex,
  Image,
  CardFooter,
  CardImageContainer,
  Tag,
  TagContainer,
  PersonImg,
  Text,
  Span,
  IconContainer,
} from './featured-listing.styles';
import { ReactComponent as BedroomIcon } from '../../assets/icons/bedroom.svg';
import { ReactComponent as WifiIcon } from '../../assets/icons/wifi.svg';
import { ReactComponent as BathtubIcon } from '../../assets/icons/bathtub.svg';
import { ReactComponent as AirConditionerIcon } from '../../assets/icons/air-conditioner.svg';
import { ReactComponent as BenchIcon } from '../../assets/icons/bench.svg';
import { ReactComponent as PinIcon } from '../../assets/icons/location-pin.svg';
import { ReactComponent as ShareIcon } from '../../assets/icons/share.svg';
import { ReactComponent as HeartIcon } from '../../assets/icons/heart.svg';

function Card({ data: info }) {
  return (
    <CardContainer>
      <CardImageContainer>
        <Image src={info.img} />

        <TagContainer>
          {info.tags.map((text) => (
            <Tag
              color={text === 'New' ? colors.orange : colors.lightgreen}
              key={text}
            >
              {text}
            </Tag>
          ))}
        </TagContainer>

        <PersonImg src={info.personImg} />
      </CardImageContainer>

      <CardBody>
        <Flex direction="column" align="flex-start">
          <Text
            size="1.125rem"
            weight="600"
            margin="0 0 0.1875rem 0"
            lineHeight="1.375rem"
          >
            {info.title}
          </Text>

          <Text
            size="0.875rem"
            margin="0 0 0.8125rem 0"
            lineHeight="1.0625rem"
            color={colors.grey}
          >
            {info.location}
          </Text>

          <Flex justify="space-between" gap="1rem">
            <Flex>
              <Span>{info.services.bedroom}</Span>
              <BedroomIcon />
            </Flex>
            <WifiIcon />
            <Flex>
              <Span>{info.services.bathroom}</Span>
              <BathtubIcon />
            </Flex>
            <AirConditionerIcon />
            <BenchIcon />
          </Flex>
        </Flex>

        <Flex gap="0.5rem">
          <PinIcon />
          <Text size="0.75rem" lineHeight="0.9375rem" color={colors.grey}>
            {info.distance}
          </Text>
        </Flex>
      </CardBody>

      <CardFooter>
        <Text
          size="1.125rem"
          weight="600"
          color={colors.black}
          lineHeight="1.375rem"
        >
          {info.price}
        </Text>

        <Flex gap="0.4375rem">
          <IconContainer iconName="heart">
            <HeartIcon />
          </IconContainer>
          <IconContainer iconName="share">
            <ShareIcon />
          </IconContainer>
        </Flex>
      </CardFooter>
    </CardContainer>
  );
}

Card.propTypes = {
  data: PropTypes.shape({
    distance: PropTypes.string,
    img: PropTypes.string,
    location: PropTypes.string,
    personImg: PropTypes.string,
    price: PropTypes.string,
    services: PropTypes.shape({
      bathroom: PropTypes.number,
      bedroom: PropTypes.number,
    }),
    tags: PropTypes.arrayOf(PropTypes.string),
    title: PropTypes.string,
  }).isRequired,
};

export default Card;
