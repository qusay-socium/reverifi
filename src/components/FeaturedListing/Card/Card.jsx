import React from 'react';
import PropTypes from 'prop-types/prop-types';

import colors from 'styles/colors';
import { ReactComponent as BedroomIcon } from 'assets/icons/bedroom.svg';
import { ReactComponent as WifiIcon } from 'assets/icons/wifi.svg';
import { ReactComponent as BathtubIcon } from 'assets/icons/bathtub.svg';
import { ReactComponent as AirConditionerIcon } from 'assets/icons/air-conditioner.svg';
import { ReactComponent as BenchIcon } from 'assets/icons/bench.svg';
import { ReactComponent as PinIcon } from 'assets/icons/location-pin.svg';
import { ReactComponent as ShareIcon } from 'assets/icons/share.svg';
import { ReactComponent as HeartIcon } from 'assets/icons/heart.svg';
import CardContainer, {
  CardBody,
  Flex,
  Image,
  CardFooter,
  CardImageContainer,
  Tag,
  TagContainer,
  PersonImg,
  TextLg,
  TextMd,
  Span,
  IconContainer,
  TextSm,
} from './card.styles';

function Card({ data: info }) {
  return (
    <CardContainer>
      <CardImageContainer>
        <Image src={info.img} />
        <TagContainer>
          {info?.tags?.map((text) => (
            <Tag
              color={text === 'New' ? colors.orange : colors.lightgreen}
              key={text}
            >
              {text}
            </Tag>
          ))}
        </TagContainer>
        <PersonImg src={info.personImg} />
      </CardImageContainer>

      <CardBody>
        <Flex direction="column" align="flex-start">
          <TextLg>{info.title}</TextLg>

          <TextMd>{info.location}</TextMd>

          <Flex justify="space-between" gap="1rem">
            <Flex>
              <Span>{info.services.bedroom}</Span>
              <BedroomIcon />
            </Flex>
            <WifiIcon />
            <Flex>
              <Span>{info?.services?.bathroom}</Span>
              <BathtubIcon />
            </Flex>
            <AirConditionerIcon />
            <BenchIcon />
          </Flex>
        </Flex>

        <Flex gap="0.5rem">
          <PinIcon />
          <TextSm>{info.distance}</TextSm>
        </Flex>
      </CardBody>

      <CardFooter>
        <TextLg>{info.price}</TextLg>

        <Flex gap="0.4375rem">
          <IconContainer iconName="heart">
            <HeartIcon />
          </IconContainer>
          <IconContainer iconName="share">
            <ShareIcon />
          </IconContainer>
        </Flex>
      </CardFooter>
    </CardContainer>
  );
}

Card.propTypes = {
  data: PropTypes.shape({
    distance: PropTypes.string,
    img: PropTypes.string,
    location: PropTypes.string,
    personImg: PropTypes.string,
    price: PropTypes.string,
    services: PropTypes.shape({
      bathroom: PropTypes.number,
      bedroom: PropTypes.number,
    }),
    tags: PropTypes.arrayOf(PropTypes.string),
    title: PropTypes.string,
  }).isRequired,
};

export default Card;
