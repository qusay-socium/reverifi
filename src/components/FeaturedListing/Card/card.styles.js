import styled from 'styled-components';

import colors from 'styles/colors';

const CardContainer = styled.div`
  background: #ffffff;
  box-shadow: 0rem 0.0625rem 0.5625rem rgba(34, 34, 34, 0.16);
  border-radius: 0.375rem;
  margin: 2rem 1rem;
  overflow: hidden;
`;

export const Flex = styled.div`
  display: flex;
  align-items: ${({ align }) => align || 'center'};
  justify-content: ${({ justify }) => justify || 'center'};
  flex-direction: ${({ direction }) => direction || 'row'};
  align-self: ${({ alignSelf }) => alignSelf || ''};
  gap: ${({ gap }) => gap || 0};
`;

export const CardImageContainer = styled.div`
  position: relative;
`;

export const Image = styled.img`
  width: 100%;
  height: 100%;
`;

export const CardBody = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  padding: 1rem;
`;

export const CardFooter = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-top: 1px solid #d8d8d8;
  padding: 0.75rem 1rem;
`;

export const TagContainer = styled.div`
  display: flex;
  position: absolute;
  top: 1rem;
  left: 1rem;
`;

export const Tag = styled.div`
  background-color: ${({ color }) => color || colors.lightgreen};
  color: white;
  font-size: 0.625rem;
  padding: 5px 1.1875rem;
  border-radius: 0.25rem;
  margin-right: 0.4375rem;
`;

export const PersonImg = styled.img`
  position: absolute;
  right: 1rem;
  bottom: 1rem;
  border-radius: 50%;
`;

export const TextLg = styled.p`
  margin: 0.1875rem 0;
  font-size: 1.125rem;
  font-weight: 600;
  color: ${colors.black};
`;

export const TextMd = styled.p`
  margin: 0.1875rem 0 0.9375rem 0;
  font-size: 0.875rem;
  color: ${colors.grey};
`;

export const TextSm = styled.p`
  margin: 0.1875rem 0;
  font-size: 0.75rem;
  color: ${colors.grey};
`;

export const Span = styled.span`
  font-size: 0.75rem;
  color: ${colors.grey};
  padding-right: 0.25rem;
`;

export const IconContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  box-shadow: 0rem 0.0625rem 0.375rem rgba(34, 34, 34, 0.163762);
  background-color: ${colors.lightgrey};
  width: 1.5625rem;
  height: 1.5625rem;
  border-radius: 50%;

  &:hover {
    background-color: ${colors.lightgreen};
    cursor: pointer;
  }

  &:hover path {
    stroke: ${({ iconName }) =>
      iconName === 'heart' ? 'white !important' : ''};
    fill: ${({ iconName }) => (iconName === 'share' ? 'white !important' : '')};
    opacity: 1 !important;
  }
`;

export default CardContainer;
