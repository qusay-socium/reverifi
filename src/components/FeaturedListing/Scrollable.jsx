import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { ScrollGrid, Dot, Dots } from './featured-listing.styles';

function Scrollable({ children, initialCount, cardsNum }) {
  const numberOfDots = Math.ceil(cardsNum / initialCount);

  const [dots, setDots] = useState(Array(numberOfDots).fill(false));

  const scrollContainer = useRef(null);

  useEffect(() => {
    setDots([true, ...dots]);
  }, []);

  const activateDot = (index) => {
    const copy = Array(numberOfDots).fill(false);
    copy[index] = true;
    setDots(copy);

    let childrenIndex = 0;

    if (!index) {
      childrenIndex = 0;
    } else if (index === numberOfDots - 1) {
      childrenIndex = cardsNum - 1;
    } else {
      childrenIndex = Math.floor(cardsNum / (numberOfDots - index)) + 1;
    }

    scrollContainer.current?.children[childrenIndex]?.scrollIntoView({
      behavior: 'smooth',
      block: 'center',
      inline: 'center',
    });
  };

  return (
    <>
      <ScrollGrid count={initialCount} ref={scrollContainer}>
        {children}
      </ScrollGrid>
      <Dots>
        {[...Array(numberOfDots)].map((_, index) => (
          <Dot active={dots[index]} onClick={() => activateDot(index)} />
        ))}
      </Dots>
    </>
  );
}

Scrollable.defaultProps = {
  cardsNum: 1,
  initialCount: 1,
};

Scrollable.propTypes = {
  cardsNum: PropTypes.number,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  initialCount: PropTypes.number,
};

export default Scrollable;
