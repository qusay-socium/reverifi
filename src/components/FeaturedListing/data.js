const data = [
  {
    distance: '1.1km away',
    img: './images/apartment1.png',
    location: 'Bronx, NY 10459',
    personImg: './images/person.png',
    price: '$123/Month',
    services: {
      bathroom: 1,
      bedroom: 2,
    },
    tags: ['New', 'Sale'],
    title: 'Apartment#1',
  },
  {
    distance: '1.1km away',
    img: './images/apartment2.png',
    location: 'Bronx, NY 10459',
    personImg: './images/person.png',
    price: '$30,000',
    services: {
      bathroom: 1,
      bedroom: 2,
    },
    tags: ['New'],
    title: 'Apartment#1',
  },
  {
    distance: '1.1km away',
    img: './images/apartment3.png',
    location: 'Bronx, NY 10459',
    personImg: './images/person.png',
    price: '$30,000',
    services: {
      bathroom: 1,
      bedroom: 2,
    },
    tags: ['Rent'],
    title: 'Apartment#1',
  },
];

export default data;
