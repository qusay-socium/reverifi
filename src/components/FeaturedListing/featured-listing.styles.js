import styled from 'styled-components';

import colors from 'styles/colors';
import mq from 'styles/media-query';

const SectionContainer = styled.div`
  margin: 0.625rem 0;

  ${mq.desktop`
      margin: 0.625rem 7.5rem;
  `}
`;

export const Title = styled.h3`
  font-weight: 600;
  font-size: 1.375rem;
  line-height: 2.4375rem;
  text-align: center;
  margin: 0;

  ${mq.mobileWide`
      font-size: 2rem;
  `}
`;

export const Flex = styled.div`
  display: flex;
  align-items: ${({ align }) => align || 'center'};
  justify-content: ${({ justify }) => justify || 'center'};
  flex-direction: ${({ direction }) => direction || 'row'};
  align-self: ${({ alignSelf }) => alignSelf || ''};
  gap: ${({ gap }) => gap || 0};
`;

export const CardImageContainer = styled.div`
  position: relative;
  height: 15.75rem;
`;

export const Image = styled.img`
  width: 100%;
  height: 100%;
`;

export const CardBody = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  padding: 1rem;
`;

export const CardFooter = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-top: 1px solid #d8d8d8;
  padding: 0.75rem 1rem;
`;

export const TagContainer = styled.div`
  display: flex;
  position: absolute;
  top: 1rem;
  left: 1rem;
`;

export const Tag = styled.div`
  background-color: ${({ color }) => color || colors.lightgreen};
  color: white;
  font-size: 0.625rem;
  padding: 0.25rem 1.1875rem;
  border-radius: 0.25rem;
  margin-right: 0.4375rem;
  line-height: 0.75rem;
`;

export const PersonImg = styled.img`
  position: absolute;
  right: 1rem;
  bottom: 1rem;
  border-radius: 50%;
`;

export const Text = styled.p`
  margin: ${({ margin }) => margin || 0};
  padding: ${({ padding }) => padding || 0};
  font-size: ${({ size }) => size || '1rem'};
  font-weight: ${({ weight }) => weight || 500};
  color: ${({ color }) => color || colors.black};
  line-height: ${({ lineHeight }) => lineHeight};
`;

export const Span = styled.span`
  font-size: 0.75rem;
  line-height: 0.9375rem;
  color: ${colors.grey};
  padding-right: 0.25rem;
`;

export const IconContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  box-shadow: 0rem 0.0625rem 0.375rem rgba(34, 34, 34, 0.163762);
  background-color: ${colors.lightgrey};
  width: 1.5625rem;
  height: 1.5625rem;
  border-radius: 50%;

  &:hover {
    background-color: ${colors.lightgreen};
    cursor: pointer;
  }

  &:hover path {
    stroke: ${({ iconName }) =>
      iconName === 'heart' ? 'white !important' : ''};
    fill: ${({ iconName }) => (iconName === 'share' ? 'white !important' : '')};
    opacity: 1 !important;
  }
`;

export const CardContainer = styled.div`
  background: #ffffff;
  box-shadow: 0rem 0.0625rem 0.5625rem rgba(34, 34, 34, 0.16);
  border-radius: 0.375rem;
  margin: 2rem 1rem;
  overflow: hidden;
`;

export const StyledSlide = styled.div`
  // dots styles
  .slideDots {
    list-style-type: none;
    text-align: center;
    margin: 0;
    padding: 0;

    > li {
      display: inline;
      margin: 0 0.3125rem;
      > button {
        // to hide the content
        font-size: 0 !important;
        width: 0.75rem;
        height: 0.75rem;
        border: none;
        background-color: #d8d8d8;
        border-radius: 50%;
        padding: 0;
      }
    }
  }
  // active dot style
  .slideDots li.slick-active button {
    width: 0.625rem;
    height: 0.625rem;
    background-color: #fff;
    outline: 2px solid #b2d235;
  }
  // arrow styles
  .slick-prev {
    left: 1rem !important;
    z-index: 1;
    width: 2rem !important;
    height: 2rem !important ;
  }
  .slick-next {
    right: 1rem !important;
    z-index: 1;
    width: 2rem !important;
    height: 2rem !important ;
  }
`;

export default SectionContainer;
