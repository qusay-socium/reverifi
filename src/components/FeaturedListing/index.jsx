import React from 'react';
import PropTypes from 'prop-types';
import Placeholder from '../../assets/placeholder.jpg';
import {
  ListingContainer,
  ListingCard,
  ListingCardHeader,
  ListingCardImage,
  ListingCardBody,
  ListingProperties,
  ListingCardFooter,
  FeatuedListingSectionContainer,
  Property,
  ListingTitleDistance,
} from './featured-listing.styles';

/**
 * Home page featured listing section.
 *
 * @return {JSX.Element}
 */
function FeaturedListing({
  imgSource,
  title,
  price,
  transactionType,
  condition,
  location,
  distance,
  icons,
}) {
  return (
    <FeatuedListingSectionContainer>
      <h2>Our Featured Listings</h2>
      <ListingContainer>
        <ListingCard>
          <ListingProperties>
            <Property condition={condition}>{condition}</Property>
            <Property>{transactionType}</Property>
          </ListingProperties>
          <ListingCardHeader>
            <ListingCardImage
              alt="Real Estate Listing"
              src={imgSource || Placeholder}
            />
          </ListingCardHeader>
          <ListingCardBody>
            <ListingTitleDistance>
              <h3>{title}</h3>
              <p>{distance}</p>
            </ListingTitleDistance>
            <h4>{location}</h4>
            <span>{icons}</span>
          </ListingCardBody>
          <ListingCardFooter>
            <p>{price}</p>
            <div>Like Share</div>
          </ListingCardFooter>
        </ListingCard>
      </ListingContainer>
      <div>
        <p>Pagination</p>
      </div>
    </FeatuedListingSectionContainer>
  );
}

FeaturedListing.propTypes = {
  condition: PropTypes.string,
  distance: PropTypes.string,
  icons: PropTypes.string,
  imgSource: PropTypes.string,
  location: PropTypes.string,
  price: PropTypes.string,
  title: PropTypes.string,
  transactionType: PropTypes.string,
};

FeaturedListing.defaultProps = {
  condition: 'New',
  distance: '1,100 km',
  icons: 'N/A',
  imgSource: '',
  location: 'Bronx, NY 12846',
  price: '$ 999 /Month',
  title: 'Apartment #1',
  transactionType: 'Sale',
};
export default FeaturedListing;
