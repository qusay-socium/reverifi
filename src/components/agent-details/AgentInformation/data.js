import { ReactComponent as Phone } from 'assets/phone.svg';
import { ReactComponent as Email } from 'assets/email.svg';
import { ReactComponent as Address } from 'assets/address.svg';
import { ReactComponent as Language } from 'assets/language.svg';
import { ReactComponent as ServiceArea } from 'assets/service-area.svg';
import { ReactComponent as Company } from 'assets/company.svg';
import { ReactComponent as Website } from 'assets/website.svg';

export const agentData = [
  {
    Icon: Phone,
    key: 'Phone',
    value: '222-222-222',
  },
  {
    Icon: Email,
    key: 'Email',
    value: 'john.dow@gmail.com',
  },
  {
    Icon: Address,
    key: 'Address',
    value: '@somewhere',
  },
  {
    Icon: Language,
    key: 'Language',
    value: 'spanish, English',
  },
  {
    Icon: ServiceArea,
    key: 'Service Area',
    value: 'New York, Alaska',
  },
];

export const aboutText =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor inc ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercita ullamco laboris nisi ut aliquip ex ea commodo consequat.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor inc ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercita ullamco laboris nisi ut aliquip ex ea commodo consequat.';

export const companyData = [
  {
    Icon: Company,
    key: 'Company',
    value: 'Company Name',
  },
  {
    Icon: Email,
    key: 'Email',
    value: 'john.doe@gmail.com',
  },
  {
    Icon: Website,
    key: 'Website',
    value: 'website.com',
  },
];

export const agentName = 'John Doe';
