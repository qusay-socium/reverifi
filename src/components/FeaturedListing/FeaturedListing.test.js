import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import colors from 'styles/colors';
import { Title, StyledSlide, Tag, Text } from './featured-listing.styles';
import FeaturedListing from './index';
import Card from './Card';
import data from './data';

describe('rendering components', () => {
  let wrapper;
  let tree;

  beforeAll(() => {
    // eslint-disable-next-line react/jsx-filename-extension
    wrapper = shallow(<FeaturedListing />);
    tree = renderer.create(<FeaturedListing />).toJSON();
  });

  it('should render FeaturedListing component without crash', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title component', () => {
    expect(wrapper.find(Title).exists()).toBeTruthy();
    expect(wrapper.find(Title).getElement().type.render.displayName).toBe(
      'styled.h3'
    );
    expect(wrapper.find(Title).type().target).toBe('h3');
    expect(wrapper.find(Title).text()).toBe('Our Featured Listing');

    // other way
    // expect(tree.children[0]).toHaveProperty('type', 'h3');
    // expect(tree.children[0].children[0]).toBe('Our Featured Listing');
  });

  it('should render slider component', () => {
    expect(wrapper.find(StyledSlide).exists()).toBeTruthy();
    expect(wrapper.find(StyledSlide).type().target).toBe('div');

    // other way
    // expect(tree.children[1]).toHaveProperty('type', 'div');
  });

  it('should render card component', () => {
    expect(wrapper.find(Card).exists()).toBeTruthy();
    expect(wrapper.find(Card).first().dive().type().target).toBe('div');
    expect(wrapper.find(Card).length).toBe(data.length);
  });
});

describe('test styled components that have props', () => {
  it('should apply default styles to Tag component', () => {
    const tree = renderer.create(<Tag />).toJSON();

    expect(tree).toHaveStyleRule('background-color', colors.lightgreen);
  });

  it('should apply props styles to Tag component', () => {
    const tree = renderer.create(<Tag color="red" />).toJSON();

    expect(tree).toHaveStyleRule('background-color', 'red');
  });

  it('should apply default styles to Text component', () => {
    const tree = renderer.create(<Text />).toJSON();

    expect(tree).toHaveStyleRule('font-size', '1rem');
  });

  it('should apply props styles to Text component', () => {
    const tree = renderer.create(<Text size="2rem" />).toJSON();

    expect(tree).toHaveStyleRule('font-size', '2rem');
  });
});
