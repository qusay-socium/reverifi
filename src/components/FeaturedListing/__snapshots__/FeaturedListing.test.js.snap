// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components should render FeaturedListing component without crash 1`] = `ShallowWrapper {}`;
